src/Microsoft.ML.AutoML/EstimatorExtensions/EstimatorExtensions.cs

```diff
@@ -256,17 +257,45 @@ public IEstimator<ITransformer> CreateInstance(MLContext context, PipelineNode p
             return CreateInstance(context, pipelineNode.InColumns[0], pipelineNode.OutColumns[0]);
         }
 
-        public static SuggestedTransform CreateSuggestedTransform(MLContext context, string inColumn, string outColumn)
+        public static SuggestedTransform CreateSuggestedTransform(MLContext context, string inColumn, string outColumn, bool sort = false)
         {
             var pipelineNode = new PipelineNode(EstimatorName.ValueToKeyMapping.ToString(),
                 PipelineNodeType.Transform, inColumn, outColumn);
+            var estimator = CreateInstance(context, inColumn, outColumn, sort);
+            return new SuggestedTransform(pipelineNode, estimator);
+        }
+
+        private static IEstimator<ITransformer> CreateInstance(MLContext context, string inColumn, string outColumn, bool sort = false)
+        {
+            if (sort)
+                return context.Transforms.Conversion.MapValueToKey(outColumn, inColumn, keyOrdinality: ValueToKeyMappingEstimator.KeyOrdinality.ByValue);
+
+            return context.Transforms.Conversion.MapValueToKey(outColumn, inColumn);
+        }
+    }
+
+    internal class ImageLoadingExtension : IEstimatorExtension
+    {
+        public static string ImageFolder { private get; set; }
```

Comment:
Continuation of https://github.com/dotnet/machinelearning/pull/4384#pullrequestreview-307343740

---

> From: @LittleLittleCloud
> Where do we Initialize ImageFolder
> 
> Or the image dataset has to be full_path?

Reply:
The most common dataset type has a relative path to the image: 
```csv
Label,SplitColumn,Weight,ImagePath
Pneumonia,Patient001,10,images/file97234.png
Pneumonia,Patient001,10,images/file09712.png
NoPneumonia,Patient002,1,images/file82193.png
NoPneumonia,Patient002,1,images/file34387.png
```

The images are then relative to the CSV/TSV dataset provided. Can you ensure this still works? The code is within the other image PR. The base path is assumed to be the CSV/TSV dataset's folder and can be overridden.

Reply:
@bigmiao.zhang@gmail.com says the dataset passed will contain full paths, we are good here.

---
In reply to: [339334860](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339334860) [](ancestors = 339334860)